application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml

```diff
@@ -271,6 +280,18 @@
       }
     }
   }
+
+  $(document).on('click', '#confirmCacheFlushModal .btn-primary', function(event) {
+    event.preventDefault();
+    const flushButton = $('#confirmCacheFlushModal .btn-primary');
+    const target = flushButton.attr('data-action');
+    $.post(target).done(function() {
+      notification = new XWiki.widgets.Notification(l10n.get('success'), 'done');
```

Comment:
You can simply use `new XWiki.widgets.Notification...`. You assign to a variable when you need to do something on the notification after (e.g. replace an already displayed notification with another one)
If you want to use it, you can display a notification for `in progress` before `post` and then replace it with the `success` one (e.g. https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/suggest/suggestAttachments.js#L451 )

Reply:
I don't think it's useful to have an `in progress`, as the operation takes very little time.